### ISSUE TITLE
Losses during training

### ISSUE THREAD

**Issue by frmrz** [david-stojanovski/echo_from_noise#2]
Hi all,
congratulations on your work! I'm trying to use the repo for one of my projects. During training, losses stabilize very quickly in about 100k steps. Is it something that you also encountered or there might be some inefficiencies in my training? thank you!

![image](https://github.com/david-stojanovski/echo_from_noise/assets/43776981/e4782647-8806-4440-af7d-e154ed329fad)

**Comment by david-stojanovski** [comment]
Hi,

So this could be completely normal. It would be something that varies with the dataset you are using, the random initialization of training and other standard parameters such as batch size and learning rate. The learning rate does get attenuated to zero over the course of the epochs so you would definitely expect a slowing of convergence the more steps you use. Further I didn't perform extensive testing on what would be an optimal amount of steps to use for training, but rather just experimented with a few different amounts of steps to see what seemed to be a reasonably safe volume of steps to use for training. I personally wouldn't be too worried about it converging fairly fast at ~100k steps, but rather that you get consistent image quality outputs.

Let me know if you have any other questions!

**Comment by frmrz** [comment]
Many thanks! The dataset I'm using has a reasonably low diversity so I can think of the model to converge quickly. I'll try to make some more runs to understand this behavior better.

One other thing I'm not getting is the setting of the number of diffusion steps. If I train with 
`__C.TRAIN.DIFFUSION_STEPS = 1000 ` 
when I'm doing inference with the DDIM sampling I still need 1000 steps to get while I was of the idea I would have been able to get good results with 100 steps or less. Do I need to set the number of steps already in the __C.TRAIN.DIFFUSION_STEPS setting?

I leave my full config below for context:
```
__C.DATASETS = edict()
__C.DATASETS.DATADIR = './data'
__C.DATASETS.SAVEDIR = './output'
__C.DATASETS.DATASET_MODE = 'myDataset'

__C.TRAIN = edict()
__C.TRAIN.DIFFUSION = edict()
__C.TRAIN.DIFFUSION.LEARN_SIGMA = True
__C.TRAIN.DIFFUSION.NOISE_SCHEDULE = "cosine"
__C.TRAIN.DIFFUSION.TIMESTEP_RESPACING = ''
__C.TRAIN.DIFFUSION.USE_KL = False
__C.TRAIN.DIFFUSION.PREDICT_XSTART = False
__C.TRAIN.DIFFUSION.RESCALE_TIMESTEPS = False
__C.TRAIN.DIFFUSION.RESCALE_LEARNED_SIGMAS = False

__C.TRAIN.IMG_SIZE = 128
__C.TRAIN.NUM_CLASSES = 6
__C.TRAIN.LR = 1e-4
__C.TRAIN.ATTENTION_RESOLUTIONS = "32,16,8"
__C.TRAIN.CHANNEL_MULT = None
__C.TRAIN.DROPOUT = 0.0
__C.TRAIN.DIFFUSION_STEPS = 1000
__C.TRAIN.SCHEDULE_SAMPLER = "uniform"
__C.TRAIN.NUM_CHANNELS = 256
__C.TRAIN.NUM_HEADS = 1
__C.TRAIN.NUM_HEADS_UPSAMPLE = -1
__C.TRAIN.NUM_HEAD_CHANNELS = 32
__C.TRAIN.NUM_RES_BLOCKS = 2
__C.TRAIN.RESBLOCK_UPDOWN = True
__C.TRAIN.USE_SCALE_SHIFT_NORM = True
__C.TRAIN.USE_CHECKPOINT = True
__C.TRAIN.CLASS_COND = False
__C.TRAIN.WEIGHT_DECAY = 0.0
__C.TRAIN.LR_ANNEAL_STEPS = 1000
__C.TRAIN.BATCH_SIZE = 1
__C.TRAIN.MICROBATCH = -1
__C.TRAIN.EMA_RATE = "0.9999"
__C.TRAIN.DROP_RATE = 0.0
__C.TRAIN.LOG_INTERVAL = 1000
__C.TRAIN.SAVE_INTERVAL = 2000
__C.TRAIN.RESUME_CHECKPOINT = False  # optional, if you want to resume training from a checkpoint
__C.TRAIN.USE_FP16 = True
__C.TRAIN.DISTRIBUTED_DATA_PARALLEL = False
__C.TRAIN.USE_NEW_ATTENTION_ORDER = True
__C.TRAIN.FP16_SCALE_GROWTH = 1e-3
__C.TRAIN.NUM_WORKERS = 16
__C.TRAIN.DETERMINISTIC = False
__C.TRAIN.NO_INSTANCE = True
__C.TRAIN.RANDOM_CROP = False
__C.TRAIN.RANDOM_FLIP = False
__C.TRAIN.IS_TRAIN = False

__C.TEST = edict()
__C.TEST.S = 1.0
__C.TEST.USE_DDIM = True
__C.TEST.DETERMINISTIC = True
__C.TEST.INFERENCE_ON_TRAIN = False
__C.TEST.BATCH_SIZE = 2
__C.TEST.CLIP_DENOISED = True
__C.TEST.NUM_SAMPLES = 100
__C.TEST.RESULTS_DIR = './output'
```

**Comment by david-stojanovski** [comment]
So you should be able to train with 1000 steps and then use less steps for the DDIM sampling. It wasn't something I explored as I wanted to remove that as a variable for the paper, but nothing has been changed for that specifically in regards to the original code. Could you try running with more DDIM steps e.g. 500 to see whether you get adequate results then. If not I can take a deeper look into the code and try a test run myself.
